### DISCUSSION TITLE
Client service blocking not working

### DISCUSSION THREAD

**Discussion by speedywrx** [hassio-addons/addon-adguard-home#211] [Q&A]
Hello, I am trying to block services on specific devices, but it doesn’t seem to be working. I have Adguard setup on my Raspberry Pi that’s running HASS and my router pointing to its static IP as the DNS server.

Steps I take:
- Settings/Client Settings
- Add device (I have tried both local IP and MAC addresses)
- under “Block Specific Services” I uncheck “use global blocked services” and check the specific services (Youtube, Facebook…)
- I restart Adguard and reboot the device and it still does not work

Any insight as to what I’m doing wrong? Everything else is working fine - it blocks ads on all my devices, but I can’t seem to get this to work.

I should also include that if I block a service across the network (Filters/Blocked Services) that does work. Just not individual clients.

**Comment by sinclairpaul** [comment]
By the sounds of it, you are using your Router as your Primary DNS and forwarding requests, so Adguard will see all requests as coming from your Router.  You need the clients to access Adguard directly.

We generally use Github for tracking Issues with the addon.  You would be better asking for how to help on the Forums/Discord.

**Comment by speedywrx** [comment]
Ok, thank you for the clarification and apologies for misunderstanding this particular forum.
